src/wallet/wallet.cpp

```diff
@@ -812,11 +813,11 @@ bool CWallet::GetAccountPubkey(CPubKey &pubKey, std::string strAccount, bool bFo
 
 void CWallet::MarkDirty()
 {
-    {
-        LOCK(cs_wallet);
-        for (std::pair<const uint256, CWalletTx>& item : mapWallet)
-            item.second.MarkDirty();
-    }
+    MarkBalancesDirty();
+
+    LOCK(cs_wallet);
```

Comment:
Am I mistaken when I think that the lock on `cs_wallet` will actually happen before the call to `MarkBalancesDirty()` due to scope? If so I would swap these lines for clarity.